sync/01-sync.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+. "$(dirname "${BASH_SOURCE[0]}")/../.common.sh"
+. "$(dirname "${BASH_SOURCE[0]}")/../.client.sh"
+
+log_message "01-sync.sh..."
+
+# 1. Trigger reconciliation for 'csvEmployee_managedUser' mapping
+RECONID=$(
+  curl -si \
+    -X POST \
+    -H "X-OpenIDM-Username: $ADMIN_USERNAME" \
+    -H "X-OpenIDM-Password: $ADMIN_PASSWORD" \
+    --connect-to "wrenidm.wrensecurity.local:80:10.0.0.11:8080" \
+    "http://wrenidm.wrensecurity.local/openidm/recon?_action=recon&mapping=csvEmployee_managedUser&waitForCompletion=true" \
+  | assert_response_status \
+  | assert_response_body '.state == "SUCCESS"' \
+  | get_response_body - \
+  | jq -r "._id"
+)
+
+# 2. Check reconciliation result
+curl -si \
+  -X GET \
+  -H "X-OpenIDM-Username: $ADMIN_USERNAME" \
+  -H "X-OpenIDM-Password: $ADMIN_PASSWORD" \
+  --connect-to "wrenidm.wrensecurity.local:80:10.0.0.11:8080" \
+  "http://wrenidm.wrensecurity.local/openidm/recon/$RECONID" \
+| assert_response_status \
+| assert_response_body '.situationSummary.ABSENT == 1' \
+| assert_response_body '.situationSummary.FOUND == 1' \
+| assert_response_body '.situationSummary.SOURCE_IGNORED == 1' \
+> /dev/null
+
+# 3. Check managed users in Wren:IDM
+curl -si \
+  -X GET \
+  -H "X-OpenIDM-Username: $ADMIN_USERNAME" \
+  -H "X-OpenIDM-Password: $ADMIN_PASSWORD" \
+  --connect-to "wrenidm.wrensecurity.local:80:10.0.0.11:8080" \
+  "http://wrenidm.wrensecurity.local/openidm/managed/user?_queryId=query-all-ids" \
+| assert_response_status \
+| assert_response_body '.resultCount == 2' \
+| assert_response_body '.result[0]._id | test("^sync(1|2)$")' \
+| assert_response_body '.result[1]._id | test("^sync(1|2)$")' \
+> /dev/null
+
+# 4. Check accounts in LDAP
+sleep 3 # Ensure that implicit synchronization has been performed
```

Comment:
This is quite suboptimal... this should be while loop or inotify on logs that checks for a message.

Reply:
Yes, I knew it. I have introduced a while loop to check the `audit/sync` logs.